components/ai_chat/common/features.cc

```diff
@@ -9,10 +9,19 @@
 
 #include "base/feature_list.h"
 #include "base/metrics/field_trial_params.h"
+#include "brave/components/ai_chat/common/buildflags/buildflags.h"
 
 namespace ai_chat::features {
 
-BASE_FEATURE(kAIChat, "AIChat", base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(kAIChat,
+             "AIChat",
+// Enable the feature flag on Android dev and nightly channels
+#if BUILDFLAG(IS_ANDROID) && BUILDFLAG(ENABLE_AI_CHAT_FEATURE_FLAG)
```

Comment:
nit: we don't need this IS_ANDROID conditional (and this comment), since the logic which determines platforms and channels to enable by default is contained in the gn file. You don't need to fix this here though, I can fix in my commit / PR

Reply:
that is handled here https://github.com/brave/brave-core/pull/20559